### PR TITLE
Added search for EAN, ISBN or query, f.e. "Harry Potter"

### PR DESCRIPTION
**Pull request by wimfeijen**
Thanks for your great work! 

You made implementing search easy.

### CODE REVIEW

**Comment by wimfeijen** [comment]
In addition, I propose to add a timeout to requests, which might be a good idea according to: http://docs.python-requests.org/en/master/user/advanced/#timeouts

Unfortunately, this ended up in the same pull request. I will try to change this.

File: bol/openapi/api.py [hunk 1]
```diff
@@ -26,6 +26,14 @@ def __init__(self, api):
     def products(self, product_ids):
         path = 'products/' + ','.join(product_ids)
         return self.request('GET', path)
+        
+    def search(self, query):
+        """
+        query might be 'Harry Potter', 'an_EAN' or 'an_ISBN'. 
+        For exact search, use extra quotation marks, for example: '"Harry Potter"'.
+        """
+        path = 'search/?q=' + query
```
Comment (pennersr) [inline bol/openapi/api.py:35]: Here, the query params are not properly URL encoded. The `self.request` needs to be extended to accept a `params` parameter... then, simply do `self.request('GET', path, params={'q': query})`

**Comment by pennersr** [comment]
I would prefer for the timeout to be user configurable, like this:

`api = OpenAPI('api_key', timeout=5)`

**Comment by pennersr** [comment]
Integrated as 2de3de9

**Pull request closed without merging**